interfaces/repo.go

```diff
@@ -764,6 +764,47 @@ func (r *Repository) connected(snapName, plugOrSlotName string) ([]*ConnRef, err
 	return conns, nil
 }
 
+// ConnectionsForDeviceKey returns all hotplug connections for given device key and interface name.
+func (r *Repository) ConnectionsForDeviceKey(deviceKey, ifaceName string) ([]*ConnRef, error) {
+	r.m.Lock()
+	defer r.m.Unlock()
+
+	snapName, err := r.guessSystemSnapName()
+	if err != nil {
+		return nil, err
+	}
+	var conns []*ConnRef
+	for _, slotInfo := range r.slots[snapName] {
+		if slotInfo.HotplugDeviceKey == deviceKey && slotInfo.Interface == ifaceName {
+			for plugInfo := range r.slotPlugs[slotInfo] {
+				connRef := NewConnRef(plugInfo, slotInfo)
+				conns = append(conns, connRef)
+			}
+		}
+	}
+
+	return conns, nil
+}
+
+// HotplugSlotForDeviceKey returns a hotplug slot for given device key and interface name or nil
+// if there is no slot.
+func (r *Repository) HotplugSlotForDeviceKey(deviceKey, ifaceName string) (*snap.SlotInfo, error) {
```

Comment:
The name should use a similar structure to the previous function, since it has exactly the same purpose and layout. So something like _SlotForHotplugKey_ instead.

The parameter also needs to be renamed and order switched, per previous note.

Doc above it needs to be changed accordingly.